### PR TITLE
fix(postgres): correctly re-acquire connection while using pg-native

### PR DESCRIPTION
**Pull request by brano543**
I have tested against local PostgreSQL database with pg-native plugin enabled.

Fixes https://github.com/sequelize/sequelize/issues/14086

### CODE REVIEW

File: src/dialects/postgres/query.js
Comment (ephys) [inline src/dialects/postgres/query.js:89]: There is no need for `new RegExp` since `/SSL SYSCALL error: EOF detected/i` is a RegExp object already

**Review by ephys — CHANGES_REQUESTED** [review]
I'm not familiar with this part of the code yet so my review is just a style review but is there an integration test we can add to check that this issue is resolved?

File: src/dialects/postgres/query.js
Comment (brano543) [reply src/dialects/postgres/query.js]: Thank you, corrected.

**Comment by brano543** [comment]
> I'm not familiar with this part of the code yet so my review is just a style review but is there an integration test we can add to check that this issue is resolved?

I am afraid I can't think of a way to test this, because this issue manifests when the underlying sockets are being closed by OS. 

I was able to replicate the issue by letting driver to connect to PostgreSQL and afterwards killing the connections with tcpkill.

**Comment by ephys** [comment]
I expected as much. It's a tricky thing to test :/

Could you add a link to this PR at the beginning of your changes? So future maintainers know to be careful when changing this code

**Comment by brano543** [comment]
@ephys Could you please merge the changes and create new release? :) This issue is breaking connectivity for us in the cloud every week. It took us months to fix this. Would be great if we could use new version from upstream instead of writing our own version of the package.

**Review by WikiRik — DISMISSED** [review]

**Comment by brano543** [comment]
@ephys 

> Merging can be performed automatically once the requested changes are addressed.
I believe it got stuck and is waiting for you to approve the change. Could you please do that so that it can get merged?

**Review by bl0cknumber — APPROVED** [review]

**Comment by brano543** [comment]
***@***.***, @WikiRik *Could you guys please remove ***@***.*** f*rom list of
reviewers? I believe pull request is pending because of that and can't be
merged. I was trying to do it,but I don't have access to do so.

št 17. 2. 2022 o 8:00 Fauzan ***@***.***> napísal(a):

**Comment by WikiRik** [comment]
I'll update the branch once again and then I will merge. Not sure when we will release a new version though, but I'll keep you in the loop

**Review by WikiRik — APPROVED** [review]

**Comment by brano543** [comment]
> I'll update the branch once again and then I will merge. Not sure when we will release a new version though, but I'll keep you in the loop

Thank you, Sir ! I hope it gets release soon. We are standing by.

**Comment by brano543** [comment]
@WikiRik Our cloud instance again got into infinite loop today because of this issue... could you please make a new release? You guys released new version, but have not included this most important fix in it - 6.16.2.

**Comment by WikiRik** [comment]
@brano543 we expect to release a new v6 release in the upcoming days including this fix

**Comment by brano543** [comment]
@WikiRik Thank you very much for an update. We are scared to go for v7 alpha release as this is a critical application for our customers. We will be looking forward to hearing back from you soon with news about including this in next v6 release!

**Comment by ephys** [comment]
I also wouldn't recommend using the V7 alpha in production yet. I'm merging a few last V7 PRs then cherry picking all of that in v6 :)

**Comment by brano543** [comment]
@ephys Thank you for your quick response. We really appreciate your work. We hope this will resolve our issues in the cloud.

**Comment by ephys** [comment]
sementic-release didn't notify on this PR because of how I included this PR in v6, but this has been included in v6.17.0 https://github.com/sequelize/sequelize/releases/tag/v6.17.0

**Comment by WikiRik** [comment]
@brano543 we've released [v6.17.0](https://github.com/sequelize/sequelize/releases/tag/v6.17.0). Let's hope this will resolve your issue

**Comment by brano543** [comment]
Hi guys, 

I did some stress test and unfortunately my fix wasn't enough if multiple users are querying the DB at the same time. I have addressed the issue fully in https://github.com/sequelize/sequelize/pull/14268. Could you please review and make a quick release again? :)

**Pull request merged**
